src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -180,4 +198,22 @@ private boolean validEdge(final AtlasObject object)
                 // Ignore any highways tagged as areas
                 && !TagPredicates.IS_AREA.test(object);
     }
+
+    /**
+     * This function checks to see if the end node of an Edge AtlasObject has an amenity tag with
+     * one of the amenityValuesToExclude.
+     * 
+     * @param object
+     *            An AtlasObject (known to be an Edge)
+     * @return {@code true} if the end node of the end has one of the amenityValuesToExclude, and
+     *         {@code false} otherwise
+     */
+    private boolean endNodeHasAmenityTypeToExclude(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+        final Optional<String> tagValue = edge.end().getTag(AmenityTag.KEY);
+
+        return tagValue.isPresent()
+                && amenityValuesToExclude.contains(tagValue.get().toUpperCase());
```

Comment:
If `AmenityTag` has the values we are looking for there, `Validators.isOfType` method could be utilized here.